---
title: "C28198 | Microsoft Docs"
ms.custom: ""
ms.date: "11/04/2016"
ms.reviewer: ""
ms.suite: ""
ms.technology: 
  - "vs-devops-test"
ms.tgt_pltfrm: ""
ms.topic: "article"
f1_keywords: 
  - "C28198"
helpviewer_keywords: 
  - "C28198"
ms.assetid: 8bad4acb-712c-43f5-81d1-45d92092d4c5
caps.latest.revision: 3
author: "corob-msft"
ms.author: "corob"
manager: "ghogen"
caps.handback.revision: 3
---
# C28198
[!INCLUDE[vs2017banner](../code-quality/includes/vs2017banner.md)]

avviso C28198: Probabile perdita di memoria a causa di un'eccezione.  
  
 L'avviso indica che la memoria allocata non viene liberata dopo che è stata sollevata un'eccezione.  L'istruzione alla fine del percorso può generare un'eccezione.  La memoria è stata passata ad una funzione, che può salvare una copia da liberare successivamente.  
  
 Questo avviso è molto simile all'avviso [C28197](../code-quality/c28197.md).  Le annotazioni che sono preferibili da utilizzare con l'avviso [C28197](../code-quality/c28197.md) possono essere utilizzate qui.  
  
## Esempio  
 Il seguente codice di esempio genera questo warning:  
  
```  
char *p1 = new char[10];  
char *p2 = new char[10];  
  
test(p1);   // does not save a copy of p  
  
delete[] p2;  
delete[] p1;  
```  
  
 L'esempio di codice seguente evita questo avviso:  
  
```  
char *p1 = new char[10];  
char *p2 = NULL;  
  
test(p1);   // does not save a copy of p  
try {  
    p2 = new char[10];  
} catch (std::bad_alloc *e) {  
    // just handle the throw  
    ;  
}  
```